nixos/modules/services/matrix/synapse.nix

```diff
@@ -759,6 +759,37 @@ in {
         ExecReload = "${pkgs.util-linux}/bin/kill -HUP $MAINPID";
         Restart = "on-failure";
         UMask = "0077";
+
+        # Security Hardening
+        # Refer to systemd.exec(5) for option descriptions.
+        CapabilityBoundingSet = [ "" ];
+        LockPersonality = true;
+        NoNewPrivileges = true;
+        PrivateDevices = true;
+        PrivateTmp = true;
+        PrivateUsers = true;
+        ProcSubset = "pid";
+        ProtectClock = true;
+        ProtectControlGroups = true;
+        ProtectHome = true;
+        ProtectHostname = true;
+        ProtectKernelLogs = true;
+        ProtectKernelModules = true;
+        ProtectKernelTunables = true;
+        ProtectProc = "invisible";
+        ProtectSystem = "strict";
+        ReadWritePaths = [ cfg.dataDir ];
+        RemoveIPC = true;
+        RestrictAddressFamilies = [ "AF_INET" "AF_INET6" "AF_UNIX" ];
+        RestrictNamespaces = true;
+        RestrictRealtime = true;
+        RestrictSUIDSGID = true;
+        SystemCallArchitectures = "native";
+        SystemCallFilter = [
+          "~@clock" "~@cpu-emulation" "~@debug"
+          "~@mount" "~@obsolete" "~@reboot"
+          "~@swap" "~@privileged" "~@resources"
+        ];
```

Comment:
How did you arrive at this list? Usually I at least allow `@system-service` explicitly before rejecting individual things.

Reply:
I took what `systemd-analyze security` suggested to exclude. Now I see that `@system-service` disables most of these as well. Do you want me to change the line to 

```nix
  SystemCallFilter = [ "@system-service" "~@resources" "~@privileged" ];
```
?

Reply:
Yeah, that looks more managable.

Reply:
Alright!